main.py

```diff
@@ -1,38 +1,69 @@
 from transformers import AutoTokenizer, AutoModelForCausalLM
 import torch
+
 # Convert datasets into a list of strings
 def load_datasets(dataset_path):
     with open(dataset_path) as f:
         text = [l.strip() for l in f if l.strip()]
         return text 
+
 # Set up the device and model
+#Use cuda if available
 device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+#Using bert-base-uncased for now: https://huggingface.co/google-bert/bert-base-uncased
 tokenizer = AutoTokenizer.from_pretrained('bert-base-uncased')
-model = AutoModelForCausalLM.from_pretrained('bert-base-uncased',output_hidden_states=True ,is_decoder=True).to(device)
+model = AutoModelForCausalLM.from_pretrained('bert-base-uncased',output_hidden_states=True,is_decoder=True).to(device)
+
+#Sets up model for use instead of training
 model.eval()
-model.to(device)
+model.to(device) #Migrate model to GPU/CPU as needed
+
 # Function to tokenize and batch the text
 def tokenize_and_batch(text , batch_size: int = 16):
+    #For all of our texts,
     for i in range(0 , len(text) , batch_size):
+        #Split each piece of text into its respective batch (e.g. 1-32, 33-64 if batch-size was 32, etc.)
         batch_texts = text[i:i+batch_size]
         batch_tokens = tokenizer(batch_texts , padding=True , truncation=True , return_tensors='pt').to(device)
-        yield batch_tokens.to(device)
+        #using yield instead of return turns this function into an iterator
+        yield batch_tokens
+
 # Function to compute means of hidden states
 def compute_means(text:list[str] , batch_size: int = 16):
-    sums , total_tokens = {} , 0
-    for batch_token in tokenize_and_batch(text , batch_size):
-        with torch.no_grad():
+    # Initialize dictionary to store layer sums and counter for total tokens (so we can divide for avg later)
+    #Note: I (farhan) switched this from a dictionary to an array, because the dictionary was being used like an array
+    #What I mean by this is that the function used keys from 0...n-1 in order as keys, and then used list comprehension afterwards to return a list anyways
+    # tldr I think gpt got a little confused and accidentally made a dictionary so it's an array now
+    sums , total_tokens = [], 0
```

Comment:
lesser known use case of typehints is if you create an empty list, then include typehint to say what kind of list is expected.

e.g. `sums: list[float] = []` or `sums: list[torch.Tensor] = []`, or whatever the type of sums should be.